From 5085a8d5dd16a3159ce43b7e071db2e1555789a0 Mon Sep 17 00:00:00 2001
From: znley <shanjiantao@loongson.cn>
Date: Thu, 16 Mar 2023 10:53:26 +0800
Subject: [PATCH] port to loong64

---
 alpine/Dockerfile | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/alpine/Dockerfile b/alpine/Dockerfile
index a88d297..85f4f90 100644
--- a/alpine/Dockerfile
+++ b/alpine/Dockerfile
@@ -133,6 +133,8 @@ RUN apk add --no-cache --virtual .build-deps \
     && echo "${RESTY_PCRE_SHA256}  pcre-${RESTY_PCRE_VERSION}.tar.gz" | shasum -a 256 --check \
     && tar xzf pcre-${RESTY_PCRE_VERSION}.tar.gz \
     && cd /tmp/pcre-${RESTY_PCRE_VERSION} \
+    && curl -L -o ./config.sub "git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=HEAD" \
+    && curl -L -o ./config.guess "git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD" \
     && ./configure \
         --prefix=/usr/local/openresty/pcre \
         --disable-cpp \
@@ -142,7 +144,7 @@ RUN apk add --no-cache --virtual .build-deps \
     && make -j${RESTY_J} \
     && make -j${RESTY_J} install \
     && cd /tmp \
-    && curl -fSL https://openresty.org/download/openresty-${RESTY_VERSION}.tar.gz -o openresty-${RESTY_VERSION}.tar.gz \
+    && curl -fSL https://github.com/Loongson-Cloud-Community/openresty/releases/download/v${RESTY_VERSION}/openresty-${RESTY_VERSION}.tar.gz -o openresty-${RESTY_VERSION}.tar.gz \
     && tar xzf openresty-${RESTY_VERSION}.tar.gz \
     && cd /tmp/openresty-${RESTY_VERSION} \
     && if [ -n "${RESTY_EVAL_POST_DOWNLOAD_PRE_CONFIGURE}" ]; then eval $(echo ${RESTY_EVAL_POST_DOWNLOAD_PRE_CONFIGURE}); fi \
-- 
2.27.0

