From 23edfdb809fecb912fdb3cce8d7fd975bed2336b Mon Sep 17 00:00:00 2001
From: yzewei <yangzewei@loongson.cn>
Date: Wed, 27 Dec 2023 09:16:54 +0800
Subject: [PATCH] add loongarch64 support

Signed-off-by: yzewei <yangzewei@loongson.cn>
---
 .../cmd/dockerfile-frontend/Dockerfile        | 26 ++++++++++---------
 1 file changed, 14 insertions(+), 12 deletions(-)

diff --git a/frontend/dockerfile/cmd/dockerfile-frontend/Dockerfile b/frontend/dockerfile/cmd/dockerfile-frontend/Dockerfile
index 23b1fd2..8d34cb6 100644
--- a/frontend/dockerfile/cmd/dockerfile-frontend/Dockerfile
+++ b/frontend/dockerfile/cmd/dockerfile-frontend/Dockerfile
@@ -1,9 +1,11 @@
-# syntax=docker/dockerfile-upstream:master
+## syntax=docker/dockerfile-upstream:master
 
 # xx is a helper for cross-compilation
-FROM --platform=$BUILDPLATFORM tonistiigi/xx:master@sha256:d4254d9739ce2de9fb88e09bdc716aa0c65f0446a2a2143399f991d71136a3d4 AS xx
+#FROM --platform=$BUILDPLATFORM tonistiigi/xx:master@sha256:d4254d9739ce2de9fb88e09bdc716aa0c65f0446a2a2143399f991d71136a3d4 AS xx
+FROM cr.loongnix.cn/tonistiigi/xx:1.3.0 AS xx
 
-FROM --platform=$BUILDPLATFORM golang:1.19-alpine AS base
+#FROM --platform=$BUILDPLATFORM golang:1.19-alpine AS base
+FROM cr.loongnix.cn/library/golang:1.19-alpine AS base
 RUN apk add git bash
 COPY --from=xx / /
 WORKDIR /src
@@ -34,15 +36,15 @@ COPY --from=build /dockerfile-frontend /bin/dockerfile-frontend
 ENTRYPOINT ["/bin/dockerfile-frontend"]
 
 
-FROM base AS buildid-check
-RUN apt-get update && apt-get --no-install-recommends install -y jq
-COPY /frontend/dockerfile/cmd/dockerfile-frontend/hack/check-daily-outdated .
-COPY --from=r.j3ss.co/reg /usr/bin/reg /bin
-COPY --from=build /dockerfile-frontend .
-ARG CHANNEL
-ARG REPO
-ARG DATE
-RUN ./check-daily-outdated $CHANNEL $REPO $DATE /out
+#FROM base AS buildid-check
+#RUN apt-get update && apt-get --no-install-recommends install -y jq
+#COPY /frontend/dockerfile/cmd/dockerfile-frontend/hack/check-daily-outdated .
+#COPY --from=r.j3ss.co/reg /usr/bin/reg /bin
+#COPY --from=build /dockerfile-frontend .
+#ARG CHANNEL
+#ARG REPO
+#ARG DATE
+#RUN ./check-daily-outdated $CHANNEL $REPO $DATE /out
 
 FROM scratch AS buildid
 COPY --from=buildid-check /out/ /
-- 
2.27.0

