From a93a48403d431240329eb6329cd6d4bb8e3dd048 Mon Sep 17 00:00:00 2001
From: zhaixiaojuan <zhaixiaojuan@loongson.cn>
Date: Tue, 28 Mar 2023 11:36:49 +0800
Subject: [PATCH 3/3] Add support for loong64

---
 Dockerfile |  7 ++++---
 Makefile   | 11 ++++++-----
 2 files changed, 10 insertions(+), 8 deletions(-)

diff --git a/Dockerfile b/Dockerfile
index 105f5bd0..56913550 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,12 +1,13 @@
 # Update the base image in Makefile when updating golang version. This has to
 # be pre-pulled in order to work on GCB.
-FROM golang:1.16.4 as build
+FROM cr.loongnix.cn/library/golang:1.19 as build
 
+RUN apt update -y && apt install -y libcap2-bin
 WORKDIR /go/src/sigs.k8s.io/metrics-server
 COPY go.mod .
 COPY go.sum .
-RUN go mod download
 
+COPY vendor vendor
 COPY pkg pkg
 COPY cmd cmd
 COPY Makefile Makefile
@@ -17,7 +18,7 @@ ARG GIT_TAG
 RUN make metrics-server
 RUN setcap cap_net_bind_service=+ep metrics-server
 
-FROM gcr.io/distroless/static:latest
+FROM cr.loongnix.cn/library/debian:buster
 COPY --from=build /go/src/sigs.k8s.io/metrics-server/metrics-server /
 USER 65534
 ENTRYPOINT ["/metrics-server"]
diff --git a/Makefile b/Makefile
index 3ecd6e86..aef60d58 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 # Common User-Settable Flags
 # --------------------------
 REGISTRY?=gcr.io/k8s-staging-metrics-server
-ARCH?=amd64
+ARCH?=loong64
 
 # Release variables
 # ------------------
@@ -11,7 +11,7 @@ BUILD_DATE:=$(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
 
 # Consts
 # ------
-ALL_ARCHITECTURES=amd64 arm arm64 ppc64le s390x
+ALL_ARCHITECTURES=amd64 arm arm64 ppc64le s390x loong64
 export DOCKER_CLI_EXPERIMENTAL=enabled
 
 # Computed variables
@@ -28,7 +28,8 @@ all: metrics-server
 # -----------
 
 SRC_DEPS=$(shell find pkg cmd -type f -name "*.go")
-CHECKSUM=$(shell md5sum $(SRC_DEPS) | md5sum | awk '{print $$1}')
+#CHECKSUM=$(shell md5sum $(SRC_DEPS) | md5sum | awk '{print $$1}')
+VERSION=0.5.0
 PKG:=k8s.io/client-go/pkg
 LDFLAGS:=-X $(PKG)/version.gitVersion=$(GIT_TAG) -X $(PKG)/version.gitCommit=$(GIT_COMMIT) -X $(PKG)/version.buildDate=$(BUILD_DATE)
 
@@ -44,8 +45,8 @@ CONTAINER_ARCH_TARGETS=$(addprefix container-,$(ALL_ARCHITECTURES))
 container:
 	# Pull base image explicitly. Keep in sync with Dockerfile, otherwise
 	# GCB builds will start failing.
-	docker pull golang:1.16.4
-	docker buildx build -t $(REGISTRY)/metrics-server-$(ARCH):$(CHECKSUM) --build-arg ARCH=$(ARCH) --build-arg GIT_TAG=$(GIT_TAG) --build-arg GIT_COMMIT=$(GIT_COMMIT) .
+	docker pull cr.loongnix.cn/library/golang:1.19
+	docker build -t $(REGISTRY)/metrics-server:$(VERSION) --build-arg ARCH=$(ARCH) --build-arg GIT_TAG=$(GIT_TAG) --build-arg GIT_COMMIT=$(GIT_COMMIT) .
 
 .PHONY: container-all
 container-all: $(CONTAINER_ARCH_TARGETS);
-- 
2.20.1

